### PR TITLE
`MvcRouteListener` initalized `onBootstrap` of the module - fix for #117

### PR DESCRIPTION
**Pull request by webimpress**
Fix for #117 with tests

### CODE REVIEW

**Comment by AdilHoumadi** [comment]
:+1:  @webimpress, Thanks for the fix!

File: src/Module.php [hunk 1]
```diff
@@ -108,4 +118,9 @@ public function onAuthenticationPost(MvcAuthEvent $e)
 
         $this->container->setService('api-identity', $e->getIdentity());
     }
+
+    public function getMvcRouteListener()
```
Comment (weierophinney) [inline src/Module.php]: Why did you add this method? It was never part of the class previously, and I don't see it used in the unit tests...

Comment (weierophinney) [inline src/Module.php]: Never mind, I *do* see it in the unit tests. That said, I think there may be a better way to do that test; I'll see if I can come up with something.

Comment (weierophinney) [inline src/Module.php]: There is; we already had the `$container` property, and can do an `assertAttributeEmpty()` on that.

Comment (webimpress) [inline src/Module.php]: @weierophinney ok, how to check these listeners:
https://github.com/zfcampus/zf-mvc-auth/pull/120/files#diff-ab83393897d6271600e706b513fd1de0R73
?

I did something similar what you have done here:
https://github.com/zfcampus/zf-versioning/commit/a9013550326ea9df432ed1b606273411a554e9a8#diff-2cf9f9b61ffec8561183510e290d9bb0R68

I don't like this solution neither but couldn't come up with anything better.

**Comment by weierophinney** [comment]
Thanks, @webimpress !

I extracted the logic in the test case for the event listeners to create a data provider instead; this ensures that if a failure occurs, we can know immediately which listener caused the issue, and also verify that other listeners are attached.

**Comment by webimpress** [comment]
@weierophinney you can remove also:
```php
$this->mvcRouteListener = 
```

https://github.com/zfcampus/zf-mvc-auth/blob/master/src/Module.php#L80
